src/main/scala/com/imaginea/activegrid/core/models/SiteFilter.scala

```diff
@@ -13,57 +11,48 @@ case class SiteFilter(override val id: Option[Long],
                       filters: List[Filter]) extends BaseEntity
 
 object SiteFilter {
-  val repository = Neo4jRepository
   val siteFilterLabel = "SiteFilter"
-  val siteFilter_AccountInfo_Rel = "HAS_ACCOUNT_INFO"
-  val siteFilter_Filters_Rel = "HAS_FILTERS"
+  val siteFilterAndAccountRelation = "HAS_ACCOUNT_INFO"
+  val siteFilterAndFiltersRelation = "HAS_FILTERS"
   val logger = LoggerFactory.getLogger(getClass)
 
   def fromNeo4jGraph(nodeId: Long): Option[SiteFilter] = {
     logger.debug(s"Executing $getClass :: fromNeo4jGraph")
-    repository.withTx {
-      neo =>
-        try {
-          val node = repository.getNodeById(nodeId)(neo)
-          if (repository.hasLabel(node, siteFilterLabel)) {
-
-            val accountAndFilter = node.getRelationships.foldLeft((AccountInfo.apply(1), List[Filter]())) {
-              (result, relationship) =>
-                val childNode = relationship.getEndNode
-                relationship.getType.name match {
-                  case `siteFilter_AccountInfo_Rel` => val accountInfo = AccountInfo.fromNeo4jGraph(childNode.getId)
-                    if (accountInfo.nonEmpty) (accountInfo.get, result._2) else result
-                  case `siteFilter_Filters_Rel` => val filter = Filter.fromNeo4jGraph(childNode.getId)
-                    if (filter.nonEmpty) (result._1, filter.get :: result._2) else result
-                }
-            }
-            Some(SiteFilter(Some(node.getId), accountAndFilter._1.asInstanceOf[AccountInfo], accountAndFilter._2))
-          } else {
-            None
+    val maybeNode = Neo4jRepository.findNodeById(nodeId)
+    maybeNode.flatMap {
+      node =>
+        if (Neo4jRepository.hasLabel(node, siteFilterLabel)) {
+          val mayBeAccountInfo = Neo4jRepository.getChildNodeId(nodeId, siteFilterAndAccountRelation).flatMap(id => AccountInfo.fromNeo4jGraph(id))
+          val accountInfo = mayBeAccountInfo match {
+            case Some(account) => account
+            case None =>
+              logger.warn(s"Account info is not found with relation $siteFilterAndAccountRelation")
+              throw new NotFoundException(s"Account info is not found with relation $siteFilterAndAccountRelation")
```

Comment:
Here AccountInfo is mandatory field.
Here we will get Option(AcoountInfo)
as we need only account info , we have to use .get on that
instead of that we are logging it and throwing exception if it None